### PR TITLE
[Metis] remove old set mpi comm process

### PR DESCRIPTION
**Pull request by philbucher**
this PR removes the old SetMPICommunicatorProcess from the MetisApp, partially because it gives some some problems when MPI is not imported first (see #4731)

I tried to update the usages in the code, but for DEM I couldn't do it bcs the Process is also a return argument of the function (I added a hint to the new syntax)

For FSI I couln't update it bcs this test still uses the deleted ModelPart Constructor

### CODE REVIEW

File: applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py [hunk 1]
```diff
@@ -82,8 +82,6 @@ def ReadModelPart(self,filename):
         model_part.AddNodalSolutionStepVariable(KratosMultiphysics.FSI_INTERFACE_RESIDUAL)
         mpi.world.barrier()
 
-        KratosMetis.SetMPICommunicatorProcess(model_part).Execute()
```
Comment (philbucher) [inline applications/FSIapplication/tests/convergence_accelerator_spring_MPI_test.py:85]: @rubenzorrilla you later use the `ParallelFillCommunicator` which also sets the MPI-comm, hence I think you can safely remove this

File: applications/TrilinosApplication/tests/test_mpi_communicator.py [hunk 2]
```diff
@@ -57,9 +57,6 @@ def _read_model_part_mpi(self,main_model_part):
         model_part_io = KratosMultiphysics.ModelPartIO(mpi_input_filename)
         model_part_io.ReadModelPart(main_model_part)
 
-        ## Construct and execute the MPICommunicator
-        KratosMetis.SetMPICommunicatorProcess(main_model_part).Execute()
-
         ## Construct and execute the Parallel fill communicator
         ParallelFillCommunicator = KratosTrilinos.ParallelFillCommunicator(main_model_part.GetRootModelPart())
```
Comment (philbucher) [inline applications/TrilinosApplication/tests/test_mpi_communicator.py:61]: same here

**Comment by philbucher** [comment]
ping @maceligueta

**Comment by maceligueta** [comment]
Hi @philbucher , I noticed that the returned value was not being used in the stage, so I just removed it. Also, MPI is not working for DEM currently, so I think we can leave the DEM with the change you proposed and the changes I made.

**Review by jcotela — APPROVED** [review]
Approving, since the remaining issue with DEM is resolved.

**Pull request merged**